### DISCUSSION TITLE
Model objective has flux but status is infeasible

### DISCUSSION THREAD

**Discussion by PkiwiBird** [opencobra/cobrapy#1266] [Q&A] [answered]
Hi,
The following is a minimal working example of what I've been experiencing:
Flux distribution of FBA creates flux through the objective but the model status is infeasible.
It's expected of this objective's reaction to have a small flux and the value correlates with experimental results.
So, I'd like to ask:
1. Is there a way to relax any parameter to allow the model to be feasible when working with such low fluxes?
   And would be correct to do that?
   I saw somewhere that is possible to change `md.solver.configuration.tolerances.optimality` or `md.solver.configuration.tolerances.feasibility ` but I'm not sure which one should I try to change and in what direction (up or down).
2. How can the solution be infeasible but we still get flux through the objective?
````
from cobra.io import load_matlab_model
from cobra import Reaction

rc_remove_lst = ['prodDNAtot']
rc_add = {'rc_id': 'prodDNAtot', 'subsystem': '', 'lb': 0.0, 'ub': float("inf"), 'mtlst': "MAM01722n:-0.009, MAM01721n:-0.991, DNAtotn:1.0"}
generic_md_name = 'prodDNAtot_w_5mC'

# load model and adapt upper bound of pool:
md_pth = 'ecModel_batch.mat'
md = load_matlab_model(md_pth)
md.reactions.get_by_id('prot_pool_exchange').upper_bound = 0.16625

# remove reaction:
md.remove_reactions(rc_remove_lst)

# add reaction to replace the one removed:
rc = Reaction(id=rc_add['rc_id'], subsystem=rc_add['subsystem'], lower_bound=rc_add['lb'], upper_bound=rc_add['ub'])
md.add_reactions([rc])
mt_d = dict()
for el in rc_add['mtlst'].split(', '):
    k, v = el.split(':')[0], el.split(':')[1]
    mt_d[k] = float(v)
r = md.reactions.get_by_id(rc_add['rc_id'])
r.add_metabolites(mt_d)

# define objective:
md.objective = 'prodDNAtot'

# add constraints on biomass:
md.reactions.get_by_id('adaptbiomass').bounds=(0.028593813, 0.030555756)

# FBA:
sol = md.optimize()
sol.status # 'infeasible'
sol.fluxes['prodDNAtot'] # 0.0007634548071
sol.objective_value # 7.122533545680217e-05
md.solver # <optlang.cplex_interface.Model at 0x7ff9cf2cdcd0>
````

base model: [ecModel_batch.zip](https://github.com/opencobra/cobrapy/files/9320915/ecModel_batch.zip)

**Comment by cdiener** [comment, marked as answer]
That was somewhat intentional to allow users to inspect a solution if it is not optimal (see https://github.com/opencobra/cobrapy/issues/688). Though, you should normally at least see a warning here. 

Though, there is a special status for numerical instability ("numeric"). You are right that increasing the feasibility tolerance would allow for slight violations, however that is often not desired. Alternatively, you could try to explicitly set `md.solver.configuration.presolve = True` which can stabilize the model.

**Reply by PkiwiBird** [reply]
it didn't solve the problem, but thank you nevertheless for the sugestion

**Reply by cdiener** [reply]
Okay, also worth checking that there aren't any constraints in the model that actually make it infeasible. Demand reactions with fixed bounds are a classic example. An easy way to check is if all your bounds include zero.
